cmd/probe/internal/highavailability/ha.go

```diff
@@ -171,6 +172,14 @@ func (ha *Ha) Start() {
 		return
 	}
 
+	isDnsReady, err := ha.IsPodReady()
+	for err != nil || !isDnsReady {
+		ha.logger.Infof("Waiting for dns resolution to be ready")
+		time.Sleep(3 * time.Second)
+		isDnsReady, err = ha.IsPodReady()
+	}
```

Comment:
why it is ready after 3 seconds?

Reply:
It's just an approximate value based on experience. Both 3 seconds or 5 seconds are acceptable.